pkg/controller/operators/olm/apiservices.go

```diff
@@ -921,3 +978,119 @@ func (a *Operator) isAPIServiceAdoptable(target *v1alpha1.ClusterServiceVersion,
 	adoptable = ownerutil.AdoptableLabels(apiService.GetLabels(), true, owners...)
 	return
 }
+
+// deleteLegacyAPIServiceResources deletes resources that were created by OLM for an APIService that used the old naming convention.
+func (a *Operator) deleteLegacyAPIServiceResources(owner ownerutil.Owner, desc v1alpha1.APIServiceDescription) error {
+	logger := log.WithFields(log.Fields{
+		"ownerName":      owner.GetName(),
+		"ownerNamespace": owner.GetNamespace(),
+		"ownerKind":      owner.GetObjectKind().GroupVersionKind().GroupKind().Kind,
+	})
+	namespace := owner.GetNamespace()
+	apiServiceName := fmt.Sprintf("%s.%s", desc.Version, desc.Group)
+
+	// Attempt to delete the legacy Service.
+	existingService, err := a.opClient.GetService(namespace, legacyAPIServiceNameToServiceName(apiServiceName))
```

Comment:
Double checking an edge case.

Given:

```go
func serviceName(deploymentName string) string {
	return deploymentName + "-service"
}
```
```go
func legacyAPIServiceNameToServiceName(apiServiceName string) string {
	return strings.Replace(apiServiceName, ".", "-", -1)
}
```

`deploymentName + "-service"` can never equal `strings.Replace(apiServiceName, ".", "-", -1)` because `apisServiceName` must be `<group>.<version>` and `<version>` cannot be `service`?

(I'm just double checking that there is no case where OLM creates "new" resources that could be detected as "legacy" resources)

Reply:
I have the same question for all of these. I wonder if it might not make sense to increment the ownership label?

Reply:
I don't believe that this is an issue. This code is:
- Called after creating / updating the APIService and related resources with the new naming convention.
- the `legacyAPIServiceNameToServiceName` and similar calls are used to get the legacy APIService resources that were created based on the old naming convention
- the `serviceName` function is used hereon for the new naming convention.

Reply:
I spoke with Evan offline to learn more about his concern and came to the conclusion that it was valid. The following code should capture the edgecase he pointed out and handle it appropriately:
```	
// Handle an edgecase where the legacy resources names matches the new names.
// This only occurs when the Group of the description matches the name of the deployment
// and the version is equal to "service".
if legacyAPIServiceNameToServiceName(apiServiceName) == serviceName(desc.DeploymentName) {
	return nil
}
```